File: app/src/main/java/com/example/studymanager/models/DTO/StudieTaskDTO.kt
package com.example.studymanager.models.DTO

import com.example.studymanager.domain.StudieTask

data class StudieTaskDTO(
    var studieTaskId: Int,
    val studieTaskTitle: String,
    val totalTaskDuration: Long,
    var remainingTaskTime: Long,
    var vakId: Int,
    var vakName: String,
    var gebruikerId: Int

) {

    fun toModel(): StudieTask {
        return StudieTask(
            studieTaskId = studieTaskId,
            studieTaskTitle = studieTaskTitle,
            totalTaskDuration = totalTaskDuration,
            remainingTaskTime = remainingTaskTime,
            vakId = vakId,
            vakName = vakName
        )
    }
}

File: Backend/StudyManagerApi/DTO/StudieTaskDTO.cs
using StudyAPI.Models.Domain;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace StudyAPI.DTO {
    public class StudieTaskDTO {

        public int studieTaskId { get; set; }

        [Required]
        public string studieTaskTitle { get; set; }

        [Required]
        public long totalTaskDuration { get; set; }

        [Required]
        public long remainingTaskTime { get; set; }

        [Required]
        public int vakId { get; set; }

        [Required]
        public string vakName { get; set; }

        [Required]
        public int gebruikerId { get; set; }

        public void UpdateFromModel(StudieTask task) {
            task.StudieTaskTitle = this.studieTaskTitle;
            task.TotalTaskDuration = this.totalTaskDuration;
            task.RemainingTaskTime = this.remainingTaskTime;
            task.VakId = this.vakId;
            task.VakName = this.vakName;
            task.GebruikerId = this.gebruikerId;
        }

    }
}
